generator/templates/go/go.jinja2

```diff
@@ -0,0 +1,723 @@
+        buffer.WriteString("    \"")
+        buffer.WriteString("{{ field.name }}")
+        buffer.WriteString("\"")
+        buffer.WriteString(": ")
+        {% if field.type.name|is_bool %}
+        buffer.WriteString(strconv.FormatBool(v.Get{{ field.name|to_go_case }}()))
+        {% else %}
+        buffer.WriteString(strconv.Itoa(int(v.Get{{ field.name|to_go_case }}())))
+        {% endif %}
+        buffer.WriteString(",")
+        buffer.WriteString("\n")
+	    {% endfor %}
+	}
+    buffer.WriteString("}")
+    buffer.WriteString("\n")
+    buffer.WriteString("}")
+	return buffer.String()
+}
+
+{% endmacro %}
+
+{% macro generate_vector(tree, archive, resource) %}
+type {{ archive.name|to_go_case }}{{ resource.name|to_go_case }}Vector struct {
+    internal flatdata.ResourceHandle
+    IsOptional bool
+    IsOpen bool
+}
+
+func (v *{{ archive.name|to_go_case }}{{ resource.name|to_go_case }}Vector) Get(i int) *{% if resource|is_multivector %}{{ resource|get_types_for_multivector(tree)|first|to_go_case }}{% else %}{{ resource|to_initializer(tree)|to_go_case }}{% endif %} {
+	return &{% if resource|is_multivector %}{{ resource|get_types_for_multivector(tree)|first|to_go_case }}{% else %}{{ resource|to_initializer(tree)|to_go_case }}{% endif %}{
+		internal: v.internal,
+		position: i,
+	}
+}
+
+func (v *{{ archive.name|to_go_case }}{{ resource.name|to_go_case }}Vector) GetSize() int {
+	size := make([]byte, 8)
+	_, err := v.internal.ReadAt(size, 0)
+	if err != nil {
+		return 0
+	}
+	intSize := binary.LittleEndian.Uint64(size)
+	return int(intSize) / {% if resource|is_multivector %}{{ resource|get_types_for_multivector(tree)|first|to_go_case(false) }}{% else %}{{ resource|to_initializer(tree)|to_go_case(false) }}{% endif %}SizeInBytes
+}
+
+func (v *{{ archive.name|to_go_case }}{{ resource.name|to_go_case }}Vector) GetSlice(start, end, step int) []*{% if resource|is_multivector %}{{ resource|get_types_for_multivector(tree)|first|to_go_case }}{% else %}{{ resource|to_initializer(tree)|to_go_case }}{% endif %} {
+	var result []*{% if resource|is_multivector %}{{ resource|get_types_for_multivector(tree)|first|to_go_case }}{% else %}{{ resource|to_initializer(tree)|to_go_case }}{% endif %}
+	
+    for start <= end {
+		result = append(result, &{% if resource|is_multivector %}{{ resource|get_types_for_multivector(tree)|first|to_go_case }}{% else %}{{ resource|to_initializer(tree)|to_go_case }}{% endif %}{
+			internal: v.internal,
+			position: start,
+		})
+		start += step
+	}
+	return result
+}
+
+func (v *{{ archive.name|to_go_case }}{{ resource.name|to_go_case }}Vector) Close() {
+    v.internal.Close()
+    v.IsOpen = false
+}
+
+func (v *{{ archive.name|to_go_case }}{{ resource.name|to_go_case }}Vector) GetSizeInBytes() int {
+    return v.internal.Len()
+}
+
+func (v *{{ archive.name|to_go_case }}{{ resource.name|to_go_case }}Vector) ToString() string {
+    buffer := bytes.Buffer{}
+    buffer.WriteString("{")
```

Comment:
Are there format strings in go?

Reply:
Sure, they exist. But I'm planning to completely change the implementation of `ToString()`. Can we submit it now with current implementation or it looks really bad?